### ISSUE TITLE
Package version don't work for upgrading package

### ISSUE THREAD

**Issue by Alex-Ji** [SharePoint/sp-dev-docs#2441]
I met a problem about package version control. Story is I have two site collections one is for demo, another is for qa test. I build a package with an increasing version number and deploy it to app catalog. all sites be affected immediately, I confirm none of sites have upgrade app by 'Get it'.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b7ee4609-11a2-b852-daed-585f9f678957
* Version Independent ID: df1d0ade-b26f-2b48-eaff-4ed2d6b1ec63
* Content: [Deploy your client-side web part to a SharePoint page (Hello World part 3)](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/serve-your-web-part-in-a-sharepoint-page)
* Content Source: [docs/spfx/web-parts/get-started/serve-your-web-part-in-a-sharepoint-page.md](https://github.com/SharePoint/sp-dev-docs/blob/master/docs/spfx/web-parts/get-started/serve-your-web-part-in-a-sharepoint-page.md)
* Product: **sharepoint**
* GitHub Login: @spdevdocs
* Microsoft Alias: **spdevdocs**

**Comment by nikskhubani** [comment]
@Alex-Ji Here is what our understanding of SPFx deployments so far;

- I created a package with version number 1.0.0.0
- Added / Deployed to App catalog
- Added to site collection or subsite
- It activated the extensions immediately
- Now I have a new version lets say 1.1.0.0
- Added / Deployed to App catalog
- Regardless of whether it's *upgraded (get it)* to the site where its added, the new extension will be deployed **_everywhere_** immediately
 
You need to plan your extension code logic accordingly. For example in our case we kept current installed version in some list and extension is loading the CDN JS as per the version specified in list instead of version coming from app catalog. 

In short, you have to manage extension version in your code instead of directly depending on the app catalog extension version

**Comment by Alex-Ji** [comment]
@nikskhubani Thank you for your reply, Finally, I found 'site collection app catalog' is a good choice for me, https://docs.microsoft.com/en-us/sharepoint/dev/general-development/site-collection-app-catalog
